webapp/src/components/admin_settings/jitsi_settings.tsx

```diff
@@ -59,235 +55,201 @@ const DEFAULT_SETTINGS: Settings = {
-                jaasapikey: DEFAULT_SETTINGS.jaasapikey,
-                jaasprivatekey: DEFAULT_SETTINGS.jaasprivatekey
-            };
-        }
-
-        this.state = {
-            mode: selectedMode,
-            value: this.props.value ?? selectedSettings
+const JitsiSettings = (props: Props) => {
+    // Check previous setting setup in case of upgrade
+    const selectedMode = props.value?.usejaas ? JAAS_MODE : JITSI_MODE;
+
+    let selectedSettings = DEFAULT_SETTINGS;
+    if (!props.value) {
+        selectedSettings = {
+            jitsiurl: props.config.PluginSettings.Plugins[pluginId].jitsiurl ?? DEFAULT_SETTINGS.jitsiurl,
+            jitsiappsecret: props.config.PluginSettings.Plugins[pluginId].jitsiappsecret ?? DEFAULT_SETTINGS.jitsiappsecret,
+            jitsiappid: props.config.PluginSettings.Plugins[pluginId].jitsiappid ?? DEFAULT_SETTINGS.jaasappid,
+            jitsicompatibilitymode: props.config.PluginSettings.Plugins[pluginId].jitsicompatibilitymode ?? DEFAULT_SETTINGS.jitsicompatibilitymode,
+            jitsiembedded: props.config.PluginSettings.Plugins[pluginId].jitsiembedded ?? DEFAULT_SETTINGS.jitsiembedded,
+            jitsijwt: props.config.PluginSettings.Plugins[pluginId].jitsijwt ?? DEFAULT_SETTINGS.jitsijwt,
+            jitsilinkvalidtime: props.config.PluginSettings.Plugins[pluginId].jitsilinkvalidtime ?? DEFAULT_SETTINGS.jitsilinkvalidtime,
+            jitsinamingscheme: props.config.PluginSettings.Plugins[pluginId].jitsinamingscheme ?? DEFAULT_SETTINGS.jitsinamingscheme,
+            usejaas: DEFAULT_SETTINGS.usejaas,
+            jaasappid: DEFAULT_SETTINGS.jaasappid,
+            jaasapikey: DEFAULT_SETTINGS.jaasapikey,
+            jaasprivatekey: DEFAULT_SETTINGS.jaasprivatekey
         };
+    }
 
-        this.onJaaSApiKeyChanged = this.onJaaSApiKeyChanged.bind(this);
-        this.onJaaSAppIDChanged = this.onJaaSAppIDChanged.bind(this);
-        this.onJaaSPrivateKeyChanged = this.onJaaSPrivateKeyChanged.bind(this);
-        this.onJaaSEmbeddedChanged = this.onJaaSEmbeddedChanged.bind(this);
-        this.onJaaSCompatibilityChange = this.onJaaSCompatibilityChange.bind(this);
+    const [value, setValue] = React.useState(props.value ?? selectedSettings);
+    const [mode, setMode] = React.useState(selectedMode);
 
-        this.onJitsiAppIDChanged = this.onJitsiAppIDChanged.bind(this);
-        this.onJitsiAppSecretChanged = this.onJitsiAppSecretChanged.bind(this);
-        this.onJitsiAuthChanged = this.onJitsiAuthChanged.bind(this);
-        this.onJitsiCompatibilityChanged = this.onJitsiCompatibilityChanged.bind(this);
-        this.onJitsiEmbeddedChanged = this.onJitsiEmbeddedChanged.bind(this);
-        this.onJitsiMeetingLinkExpChanged = this.onJitsiMeetingLinkExpChanged.bind(this);
-        this.onJitsiMeetingNamesChanged = this.onJitsiMeetingNamesChanged.bind(this);
-        this.onJitsiURLChanged = this.onJitsiURLChanged.bind(this);
-    }
+    React.useEffect(() => {
```

Comment:
```suggestion
    useEffect(() => {
```
Extract this from React at the top.